File: 1346-check-if-n-and-its-double-exist/1346-check-if-n-and-its-double-exist.java
class Solution {
    public boolean checkIfExist(int[] arr) {
        HashSet<Integer> hashSet = new HashSet<>();

        for(int i : arr){
            if(hashSet.contains(i*2) || (hashSet.contains(i/2) && i%2==0)){
                return true;
            }
            hashSet.add(i);
        }

        return false;
    }
}

File: 1346-check-if-n-and-its-double-exist/1346-check-if-n-and-its-double-exist.kt
class Solution {
    fun checkIfExist(arr: IntArray): Boolean {
        
            
       for(i in arr.indices) {

            for(j in i+1 until arr.size) {

                if(arr[i] == 2 * arr[j] || arr[j] == 2 * arr[i]) {
                    return true
                }
            }
        }
        
        return false
    }
}